client/devdocs/main.jsx

```diff
@@ -130,17 +130,17 @@ module.exports = React.createClass( {
 
 	snippet: function( result ) {
 		// split around <mark> tags to avoid setting unescaped inner HTML
-		var parts = result.snippet.split(/(<mark>.*?<\/mark>)/);
+		var parts = result.snippet.split( /(<mark>.*?<\/mark>)/ );
 
 		return (
 			<div className="devdocs__result-snippet" key={ 'snippet' + result.path }>
 				{ parts.map( function( part, i ) {
 					var markMatch = part.match( /<mark>(.*?)<\/mark>/ );
 					if ( markMatch ) {
-						return <mark key={ 'mark' + i }>{markMatch[1]}</mark>;
-					} else {
-						return part;
+						return <mark key={ 'mark' + i }>{ markMatch[1] }</mark>;
```

Comment:
Currently it is not required because we're returning html. I think we should escape the html on the server side and then decodeEntities on the client side. What say?